tests/test_string_accessors.py

```diff
@@ -0,0 +1,398 @@
+    _check(assert_type(idx.str.rsplit("a", expand=True), pd.MultiIndex))
+    _check(assert_type(idx.str.split("a", expand=True), pd.MultiIndex))
+
+    # These ones are the odd ones out?
+    check(assert_type(idx.str.extractall(r"([ab])?(\d)"), pd.DataFrame), pd.DataFrame)
+    check(assert_type(idx.str.extract(r"([ab])?(\d)"), pd.DataFrame), pd.DataFrame)
+
+
+def test_series_overloads_partition():
+    s = pd.Series(
+        [
+            "ap;pl;ep",
+            "ban;an;ap",
+            "Che;rr;yp",
+            "DA;TEp",
+            "eGGp;LANT;p",
+            "12;3p",
+            "23.45p",
+        ]
+    )
+    check(assert_type(s.str.partition(sep=";"), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(s.str.partition(sep=";", expand=True), pd.DataFrame), pd.DataFrame
+    )
+    check(
+        assert_type(s.str.partition(sep=";", expand=False), "pd.Series[type[object]]"),
+        pd.Series,
+        object,
+    )
+
+    check(assert_type(s.str.rpartition(sep=";"), pd.DataFrame), pd.DataFrame)
+    check(
+        assert_type(s.str.rpartition(sep=";", expand=True), pd.DataFrame), pd.DataFrame
+    )
+    check(
+        assert_type(s.str.rpartition(sep=";", expand=False), "pd.Series[type[object]]"),
+        pd.Series,
+        object,
+    )
+
+
+def test_index_overloads_partition():
+    idx = pd.Index(
+        [
+            "ap;pl;ep",
+            "ban;an;ap",
+            "Che;rr;yp",
+            "DA;TEp",
+            "eGGp;LANT;p",
+            "12;3p",
+            "23.45p",
+        ]
+    )
+    check(assert_type(idx.str.partition(sep=";"), pd.MultiIndex), pd.MultiIndex)
+    check(
+        assert_type(idx.str.partition(sep=";", expand=True), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(idx.str.partition(sep=";", expand=False), "pd.Index[type[object]]"),
+        pd.Index,
+        object,
+    )
+
+    check(assert_type(idx.str.rpartition(sep=";"), pd.MultiIndex), pd.MultiIndex)
+    check(
+        assert_type(idx.str.rpartition(sep=";", expand=True), pd.MultiIndex),
+        pd.MultiIndex,
+    )
+    check(
+        assert_type(
+            idx.str.rpartition(sep=";", expand=False), "pd.Index[type[object]]"
+        ),
+        pd.Index,
+        object,
+    )
+
+
+def test_series_overloads_cat():
+    s = pd.Series(DATA)
+    check(assert_type(s.str.cat(sep=";"), str), str)
+    check(assert_type(s.str.cat(None, sep=";"), str), str)
+    check(
+        assert_type(
+            s.str.cat(["A", "B", "C", "D", "E", "F", "G"], sep=";"),
+            "pd.Series[str]",
+        ),
+        pd.Series,
+        str,
+    )
+    check(
+        assert_type(
+            s.str.cat(pd.Series(["A", "B", "C", "D", "E", "F", "G"]), sep=";"),
+            "pd.Series[str]",
+        ),
+        pd.Series,
+        str,
+    )
+    unknown_s = pd.DataFrame({"a": list("abcdefg")})["a"]
+    check(assert_type(s.str.cat(unknown_s, sep=";"), "pd.Series[str]"), pd.Series, str)
```

Comment:
I think you should also test things like:
```python
check(assert_type(unknown_s.str.cat(s, sep=";"), "pd.Series[str]"), pd.Series, str)
check(assert_type(unknown_s.str.cat(unknown_s, sep=";"), "pd.Series[str]"), pd.Series, str)
```
Ideally, we should do this for all of the methods to make sure that when an `UnknownSeries` or `UnknownIndex` is passed, then we get an `pd.Series[str]` and `Index[str]` respectively.